src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
+
+        public override string ConvertName(string name)
+        {
+            int bufferLength = name.Length * 2;
+            char[]? buffer = bufferLength > 512
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
+                : buffer;
```

Comment:
For consistency, please use the shared threshold for char allocations (currently set at 128 chars):
```suggestion
            char[]? buffer = bufferLength > JsonConstants.StackallocCharThreshold
                ? ArrayPool<char>.Shared.Rent(bufferLength)
                : null;

            int resultLength = 0;
            Span<char> result = buffer is null
                ? stackalloc char[JsonConstants.StackallocCharThreshold]
                : buffer;
```